tests/system_tests_link_routes.py

```diff
@@ -1797,110 +1814,227 @@ def _fake_broker(self, cls):
         self.router.wait_connectors()
         return fake_broker
 
+    def _find_frame(self, data, code) -> Optional[list]:
+        """Scan a byte sequence for performatives that match code.
+        Return the frame body (list) if match else None
+        """
+        while data:
+            # starts at frame header (8 bytes)
+            frame_len = int.from_bytes(data[:4], "big")
+            if frame_len == 0 or frame_len > len(data):
+                return None
+            desc = Data()
+            desc.decode(data[8:frame_len])  # skip frame header
+            data = data[frame_len:]    # advance to next frame
+            desc.rewind()
+            if desc.next() is None:
+                return None
+            if not desc.is_described():
+                return None
+            py_desc = desc.get_py_described()
+            if py_desc.descriptor == code:
+                return py_desc.value
+        return None
+
+    def _send_frame(self, frame: Data, sock: socket.socket):
+        """Encode and send frame over sock
+        """
+        frame.rewind()
+        fbytes = frame.encode()
+        flen = len(fbytes) + 8
+        # AMQP FRAME HEADER: 4 byte length, DOFF, TYPE, CHANNEL
+        sock.sendall(flen.to_bytes(4, "big"))
+        sock.sendall(bytes([2, 0, 0, 0]))
+        sock.sendall(fbytes)
+
+    def _construct_transfer(self, delivery_id, tag, more=False, add_ma=False,
```

Comment:
I can imagine that these functions will in the future need to be moved up the scope so that other tests can use them too. They look a bit complicated, maybe even enough to deserve their own test, maybe just a small one?